internal/vision/provider_test.go

```
@@ -0,0 +1,294 @@
+		s.gotMethod = r.Method
+		s.gotPath = r.URL.Path
+		s.auth = r.Header.Get("Authorization")
+		b, _ := io.ReadAll(r.Body)
+		s.body = b
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(s.status)
+		if s.status != http.StatusOK {
+			_, _ = io.WriteString(w, `{"error":{"message":"injected failure"}}`)
+			return
+		}
+		fmt.Fprintf(w, `{"id":"1","object":"chat.completion","choices":%s}`, s.choices)
```

golangci-lint (2.12.2)

[error] Error return value of `fmt.Fprintf` is not checked

(errcheck)